### DISCUSSION TITLE
How to load huge dataset for conversion into spacy JSON in colab

### DISCUSSION THREAD

**Discussion by Parth-tech** [explosion/spaCy#8506] [Help: Coding & Implementations] [answered]
I have a dataset having 65 million entries in IOB format. 
So, while converting it to spacy Binary format using the command:

_**!python -m spacy convert /content/train.tsv /content/ -s -c ner**_

The colab would show the:
_**tcmalloc: large alloc .....**_
**_^C_**

Could you please give me some way of circumventing this issue. I don't have a lot of experience with colab or profound knowledge of data science concepts. Any reference would also work.

Is it possible to make the .spacy file in increments? 

Thank you!

**Comment by polm** [comment, marked as answer]
You can split your IOB file into multiple files and then convert them individually into `.spacy` files. When specifying data in the training config, you can give a directory full of `.spacy` or `.json` files and it'll work the same as a single file.

**Reply by Parth-tech** [reply]
Oh Thank you so much for such a quick response. 

One last thing @polm if I set path to the directory having multiple .spacy, then, while training the model would it load the files one at a time or all in one go?

The RAM wouldn't overflow while training right sir?

**Reply by polm** [reply]
Training data is generally all read in to facilitate batch shuffling. You should looks at the docs on [custom data reading](https://spacy.io/usage/training#custom-code-readers-batchers), which includes an explanation of how to stream in your training data rather than getting it all in memory at once.

**Reply by Parth-tech** [reply]
Thank you so much @polm. Great help!
